severach commented on 2023-03-12 22:02 (UTC)
The update of third-party/{pjproject,versions.mak} from 2.12.1 to 2.13 is detecting libupnp on the system, usually from vlc, found on desktop systems, not found on Asterisk servers. This changes pjnath to -DPJNATH_HAS_UPNP=1 which causes -lupnp -lixml. Fix with this patch.

diff --color -ru asterisk-20.2.0_vanilla/third-party/pjproject/Makefile.rules asterisk-20.2.0_libupnp-fix/third-party/pjproject/Makefile.rules
--- asterisk-20.2.0_vanilla/third-party/pjproject/Makefile.rules	2023-03-09 17:17:03.000000000 +0000
+++ asterisk-20.2.0_libupnp-fix/third-party/pjproject/Makefile.rules	2023-03-13 14:49:56.566284230 +0000
@@ -15,6 +15,7 @@
 # Even though we're not installing pjproject, we're setting prefix to /opt/pjproject to be safe
 
 PJPROJECT_CONFIG_OPTS = $(PJPROJECT_CONFIGURE_OPTS) --prefix=/opt/pjproject \
+	--disable-upnp \
 	--disable-speex-codec \
 	--disable-speex-aec \
 	--disable-bcg729 \
